XMonad/Prompt/OrgMode.hs

```diff
@@ -0,0 +1,391 @@
+    -- * Usage
+    -- $usage
+
+    -- * Prompts
+    orgPrompt,              -- :: XPConfig -> String -> FilePath -> X ()
+    orgPromptPrimary,       -- :: XPConfig -> String -> FilePath -> X ()
+
+    -- * Types
+    ClipboardSupport (..),
+    OrgMode,                -- abstract
+) where
+
+import XMonad (X, io)
+import XMonad.Prompt (XPConfig, XPrompt (showXPrompt), mkXPrompt)
+import XMonad.Util.XSelection (getSelection)
+
+import Control.Applicative (empty)
+import Control.Monad ((<=<))
+import Data.Char (isDigit)
+import Data.Functor ((<&>))
+import Data.List (foldl')
+import Data.Maybe (fromMaybe, listToMaybe)
+import Data.Time (Day (ModifiedJulianDay), NominalDiffTime, UTCTime (utctDay), addUTCTime, defaultTimeLocale, formatTime, fromGregorian, getCurrentTime, iso8601DateFormat, nominalDay, toGregorian)
+import System.Directory (getHomeDirectory)
+import System.IO (IOMode (AppendMode), hPutStrLn, withFile)
+import Text.ParserCombinators.ReadP (ReadP, munch, munch1, readP_to_S, skipSpaces, string, (<++))
+
+{- $usage
+
+You can use this module by importing it, along with "XMonad.Prompt", in
+your @xmonad.hs@
+
+> import XMonad.Prompt
+> import XMonad.Prompt.OrgMode (orgPrompt)
+
+and adding an appropriate keybinding.  For example, using syntax from
+"XMonad.Util.EZConfig":
+
+> , ("M-C-o", orgPrompt def "TODO" "/home/me/org/todos.org")
+
+This would create notes of the form @* TODO /my-message/@ in the
+specified file.
+
+You can also enter a relative path; in that case the file path will be
+prepended with @$HOME@ or an equivalent directory.  I.e. instead of the
+above you can write
+
+> , ("M-C-o", orgPrompt def "TODO" "org/todos.org")
+
+There is also some scheduling and deadline functionality present.  They
+are initiated by entering @+s@ or @+d@ into the prompt respectively,
+followed by a date and a time of day.  Any of the following are valid
+dates:
+
+  - today
+  - tomorrow
+  - /any weekday/
+  - /any date of the form DD MM YYYY/
+
+In the last case, the month and the year are optional and will be, if
+missing, filled out with the current month and year.  We disambiguate as
+early as possible, so a simple @w@ will suffice to mean Wednesday, while
+@s@ will not be enough to say Sunday.  Weekdays also always schedule
+into the future, e.g. if today is Monday and you schedule something for
+Monday, you will actually schedule it for the /next/ Monday (the one in
+seven days).
+
+The time is specified in the @HH:MM@ format.  The minutes may be
+omitted, in which case @00@ will be substituted.
+
+A few examples are probably in order.  Suppose we have bound the key
+above, pressed it, and are now confronted with a prompt:
+
+  - @hello +s today@ would create a TODO note with the header @hello@
+    and would schedule that for today's date.
+
+  - @hello +d today 12@ schedules the note for today at 12:00.
+
+  - @hello +d today 12:30@ schedules it for today at 12:30.
+
+  - @hello +d today 12:30@ works just like above, but creates a
```

Comment:
I'm not sure how it's different from the above ? Even the above one seems to be a deadline ?

Reply:
Whoops, you're right.  The two above should read `+s` instead of `+d`.  That way "schedules" also makes sense :)